pkg/ext-proc/backend/datastore.go

```diff
@@ -9,24 +10,65 @@ import (
 	"k8s.io/klog/v2"
 )
 
+func NewK8sDataStore(options ...K8sDatastoreOption) *K8sDatastore {
+	store := &K8sDatastore{
+		poolMu:      sync.RWMutex{},
+		llmServices: &sync.Map{},
+		pods:        &sync.Map{},
+	}
+	for _, opt := range options {
+		opt(store)
+	}
+	return store
+}
+
 // The datastore is a local cache of relevant data for the given LLMServerPool (currently all pulled from k8s-api)
 type K8sDatastore struct {
-	LLMServerPool *v1alpha1.LLMServerPool
-	LLMServices   *sync.Map
-	Pods          *sync.Map
+	// poolMu is used to synchronize access to the llmServerPool.
+	poolMu        sync.RWMutex
+	llmServerPool *v1alpha1.LLMServerPool
+	llmServices   *sync.Map
+	pods          *sync.Map
+}
+
+type K8sDatastoreOption func(*K8sDatastore)
+
+// WithPods can be used in tests to override the pods.
+func WithPods(pods []*PodMetrics) K8sDatastoreOption {
+	return func(store *K8sDatastore) {
+		store.pods = &sync.Map{}
+		for _, pod := range pods {
+			store.pods.Store(pod.Pod, true)
+		}
+	}
+}
+
+func (ds *K8sDatastore) SetLLMServerPool(pool *v1alpha1.LLMServerPool) {
+	ds.poolMu.Lock()
```

Comment:
have we experienced/do we expect a race condition with the serverPool?

Reply:
yes, happens pretty often actually. If you try restarting EPP many times, you can probably hit it

Reply:
Yes, I agree that the startup race condition is annoying, and this PR resolves that in your predicate func. But I think this is unrelated. I suppose its not a big deal, and can keep us safer. Getters/Setters can just add code bloat and can complicate simple logic. Tying this into the export comment thread as well based on the same idea.

Reply:
Yeah it's common practice to hide internal fields until it's necessary to be exported, it reduces chances of bugs. The llmServerPool field is particular important here because I need to synchronize the access to it.

Reply:
Yeah, I think thats more Java/C# style, but I won't dig too hard here.

Reply:
So go doesn't encourage the "GetXX" naming, so in this case I should name it "LLMServerPool" instead of "GetLLMServerPool" But Getter/Setter pattern is common in Go as well, see the "Getter" section https://go.dev/doc/effective_go